cmd/maya-apiserver/cstor-operator/spc/handler.go

```diff
@@ -121,6 +138,111 @@ func (c *Controller) syncSpc(spc *apis.StoragePoolClaim) error {
 	return nil
 }
 
+// handleSPCDeletion handles deletion of a SPC resource by deleting
+// the associated CSP resource to it, removing the SPC finalizer
+// on BDC(s) used and then removing the SPC finalizer on SPC resource
+// itself.
+
+// It is necessary that SPC resource has the SPC finalizer on it in order to
+// execute the handler.
+func handleSPCDeletion(spc *apis.StoragePoolClaim) error {
+	err := deleteAssociatedCSP(spc)
+
+	if err != nil {
+		return errors.Wrapf(err, "failed to handle spc deletion")
+	}
+
+	if spcv1alpha1.BuilderForAPIObject(spc).Spc.HasFinalizer(spcv1alpha1.SPCFinalizer) {
+		err := removeSPCFinalizer(spc)
+		if err != nil {
+			return errors.Wrapf(err, "failed to handle spc deletion")
+		}
+	}
+
+	return nil
+}
+
+// deleteAssociatedCSP deletes the CSP resource(s) belonging to the given SPC resource.
+// If no CSP resource exists for the SPC, then a levelled info log is logged and function
+// returns.
+func deleteAssociatedCSP(spc *apis.StoragePoolClaim) error {
+	err := csp.KubeClient().DeleteCollection(
+		metav1.ListOptions{
+			LabelSelector: string(apis.StoragePoolClaimCPK) + "=" + spc.Name,
+		},
+		&metav1.DeleteOptions{},
+	)
+
+	if k8serror.IsNotFound(err) {
+		glog.V(2).Infof("Associated CSP(s) of storagepoolclaim %s is already deleted:%s", spc.Name, err.Error())
+		return nil
+	}
+
+	if err != nil {
+		return errors.Wrapf(err, "Failed to delete associated CSP(s):%s", err.Error())
+	}
+
+	return nil
```

Comment:
IMO here we need to return error.  So next reconciliation time nil will be returned from k8serror.IsNotfound.

Reply:
Got to see in below function not required